cylc/flow/config.py

```diff
@@ -206,9 +214,9 @@ def __init__(
             'first-parent descendants': {},
         }
         # tasks
-        self.leaves = []
+        self.leaves = []  # TODO figure out type
```

Comment:
Are you planning to address these TODOs on this PR?

Reply:
Not really, unless someone can tell me what the type is supposed to be (I struggled to figure it out in a few minutes, and I'm not sure if it's worth spending more time on it until the code needs changing)